external_accounts/views.py

```diff
@@ -157,4 +159,51 @@ def citesphere_disconnect(request, repository_id):
             'message': 'Error disconnecting Citesphere account. Please try again or contact support if the issue persists.'
         })
 
+    return redirect(reverse('dashboard'))
+
+@login_required
+def conceptpower_login(request):
+    if request.method == 'POST':
+        username = request.POST.get('username')
+        password = request.POST.get('password')
+        next_url = request.POST.get('next', reverse('dashboard'))
+
+        if not username or not password:
+            messages.error(request, "Both username and password are required.")
+            return render(request, 'login/login.html')
+
+        account, created = ConceptpowerAccount.objects.get_or_create(user=request.user)
+        account.username = username
+        account.password = password
```

Comment:
password should not be stored in plaintext. it should be encrypted in a way that we can get the original password back, but at least that.